scala-package/core/src/main/scala/ml/dmlc/mxnet/Optimizer.scala

```diff
@@ -139,4 +191,6 @@ trait MXKVStoreUpdater {
    */
   def update(key: Int, recv: NDArray, local: NDArray): Unit
   def dispose(): Unit
+  def serializeState(): Array[Byte]
+  def deserializeState(bytes: Array[Byte]): Unit
```

Comment:
In my opinion, it's better to have the consistent API signature in `MXKVStoreUpdater ` and `Optimizer `.....

but from the implementation, it seems that you make an imperative implementation which updates the internal state of MXKVStoreUpdater, which is a `val states = new scala.collection.mutable.HashMap[Int, AnyRef]`....In this case, shall we make `val states = new scala.collection.mutable.HashMap[Int, AnyRef]` in trait and add comments in deserializeState() telling the developer that, you are supposed to update this data structure in this method?